### PR TITLE
Two crash handler improvements

### LINKED ISSUE #32537: systemd.crash_shell runs in tandem with serial-getty@ttyS0 causing issues
### systemd version the issue has been seen with

main

### Used distribution

Fedora 39

### Linux kernel version used

_No response_

### CPU architectures issue was seen on

None

### Component

_No response_

### Expected behaviour you didn't see

When PID 1 crashes and `systemd.crash_shell` is enabled, the spawned terminal takes control of the /dev/console TTY. 

### Unexpected behaviour you saw

When PID 1 crashes and `systemd.crash_shell` is enabled, both the spawned crash shell and `serial-getty@ttyS0` are reading from /dev/console, leading to an unusable terminal as both try to read from it when characters are entered.

### Steps to reproduce the problem

```
mkosi -f qemu
kill -s ABRT 1
```

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by daandemeyer**
Fixes #32537

### CODE REVIEW

File: src/core/crash-handler.c [hunk 1]
```diff
@@ -155,16 +155,14 @@ _noreturn_ static void crash(int sig, siginfo_t *siginfo, void *context) {
         (void) sigaction(SIGCHLD, &sa, NULL);
 
         if (arg_crash_shell) {
-                log_notice("Executing crash shell in 10s...");
```
Comment (YHNdnzj) [inline src/core/crash-handler.c:158]: Please update the documentation.

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
I gather you are having some fun?

**Comment by DaanDeMeyer** [comment]
> I gather you are having some fun?

Depends on your definition of fun. Good thing is we're very close to having debugging not be a PITA anymore.

File: mkosi.extra/root/.gdbinit [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+set debuginfod enabled off
```
Comment (bluca) [inline mkosi.extra/root/.gdbinit:3]: if you are adding this file, how about setting the source code dir as well?

Comment (daandemeyer) [reply mkosi.extra/root/.gdbinit:3]: Not required it's baked into the debuginfo

File: src/core/crash-handler.c [hunk 3]
```diff
@@ -155,9 +155,6 @@ _noreturn_ static void crash(int sig, siginfo_t *siginfo, void *context) {
...
-                (void) sleep(10);
```
Comment (poettering) [inline src/core/crash-handler.c:159]: i am fine with dropping the sleeping here. But I think we should keep a message here that is shown ebfore we fork off the crash shell, just to make clear that this is what happens now.

i.e. keep `log_notice("Executing crash shell...");`

in place

**Pull request merged**